patches/minecraft/net/minecraft/world/chunk/Chunk.java.patch

```diff
@@ -299,4 +312,29 @@
 +            }
 +        }
 +    }
++
++    private static ThreadLocal<Boolean> populating = new ThreadLocal<Boolean>(); // keep track of cascading chunk generation during chunk population
++
++    /** for Forge internal use, to detect cascading chunk generation issues */
++    private static boolean isPopulatingChunk()
++    {
++        return populating.get() == Boolean.TRUE;
++    }
++
++    /** for Forge internal use, to detect cascading chunk generation issues */
++    private static void setPopulatingChunk(Boolean populatingTerrain)
```

Comment:
Shouldn't this just take a `boolean` parameter, unless there's a reason to set `null`?